src/os/windows/win32.cpp

```diff
@@ -669,13 +669,13 @@ int OTTDStringCompare(const char *s1, const char *s2)
 		int len_s2 = MultiByteToWideChar(CP_UTF8, 0, s2, -1, nullptr, 0);
 
 		if (len_s1 != 0 && len_s2 != 0) {
-			LPWSTR str_s1 = AllocaM(WCHAR, len_s1);
-			LPWSTR str_s2 = AllocaM(WCHAR, len_s2);
+			std::wstring str_s1(len_s1 + 1, L'\0');
+			std::wstring str_s2(len_s2 + 1, L'\0');
```

Comment:
```suggestion
			std::wstring str_s1(len_s1, L'\0');
			std::wstring str_s2(len_s2, L'\0');
```
MultiByteToWideChar() returned length includes terminating null character, so not using +1 was fine in old code.